population collection map

cclarkson opened the pull request:
Hi @alimanfoo, @hardingnj,
Here's a first pass at a phase two map that I think is easily scaled up to a phase three map.
Thoughts?

Resolves #9.

alimanfoo commented:
Hi Chris, looks great. I think the general design is a good approach to dealing with more populations. I also think its fine to hide the details of multiple collection sites per population, that is a detail we could always provide in supplementary information.

Some minor suggestions:

* Change the population colours to use the [colour scheme](https://github.com/malariagen/agam-report-base/blob/f30f89d829eb2bd302b868dd79fe4840b34cd322/src/python/ag1k/phase2_ar1.py#L57) in the agam-report-base phase2_ar1.py module.
* Make the legend title just "Populations", then change population labels to e.g. "Ghana coluzzii (n=55)".
* For countries where there are both gambiae and coluzzii populations, choose a single location for both populations. E.g., currently GHcol and GHgam have pointers to slightly different locations, but these should be the same. If you want to find a location e.g. for both GH populations, then take the average coordinates for all GH samples.

One other point for discussion, I think it might be nice to put the ecosystem classifications as the base layer, but happy to discuss if that starts to clash with population colours. Alternatively I think we need a base layer with higher resolution, the blockiness currently doesn't look great.

Thanks.

alimanfoo commented:
In the legend, also suggest:
* Change "Equatorial Guinea gambiae" to "Bioko island gambiae" (or maybe just "Bioko gambiae")
* Change "French Mayotte gambiae" to "Mayotte island gambiae" (or maybe just "Mayotte gambiae")

cclarkson commented:
Thanks @alimanfoo,

I'd forgotten we had phase 2 colours :+1: 

I've addressed all your points apart from the base layer in the last commit. I'm going to have a play with making the map look higher resolution now.

Cheers

alimanfoo commented:
Thanks Chris, looking good.

Btw if you want to play with bringing in the African ecosystems, there is
code here for doing it with cartopy:
http://alimanfoo.github.io/2016/10/04/plotting-african-ecosystems.html

On Fri, 16 Nov 2018 at 14:53, Chris Clarkson <notifications@github.com>
wrote:

> Thanks @alimanfoo <https://github.com/alimanfoo>,
>
> I'd forgotten we had phase 2 colours 👍
>
> I've addressed all your points apart from the base layer in the last
> commit. I'm going to have a play with making the map look higher resolution
> now.
>
> Cheers
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/malariagen/ag1000g-phase2-data-paper/pull/10#issuecomment-439417387>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq8QgDNByRQuvHZTU_5cVCJI5EbR1_mks5uvtF6gaJpZM4YmDYS>
> .
>


-- 
Please feel free to resend your email and/or contact me by other means if
you need an urgent reply.

Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health
Big Data Institute
Li Ka Shing Centre for Health Information and Discovery
Old Road Campus
Headington
Oxford
OX3 7LF
United Kingdom
Phone: +44 (0)1865 743596 or +44 (0)7866 541624
Email: alimanfoo@googlemail.com
Web: http://a <http://purl.org/net/aliman>limanfoo.github.io/
Twitter: @alimanfoo <https://twitter.com/alimanfoo>

cclarkson commented:
@alimanfoo, @hardingnj 

If either of you can install/import and run gdal e.g in the example from `http://alimanfoo.github.io/2016/10/04/plotting-african-ecosystems.html`, I will give you 50p.

I think that conda might have broken it again, see https://github.com/ContinuumIO/anaconda-issues/issues/9449 and I have left a new issue over there but it would be cool if either of you could give it a quick test just to confirm.

cclarkson commented:
Hi @alimanfoo, @hardingnj, @EricRLucas 

This is ready for another gander, I got the ecological zone GIS data working.

alimanfoo commented:
Thanks Chris, looks great I think with the ecosystems background layer.

Would it be possible to group and order the populations within the legend, so the blues are together, the reds are together, etc., and within each colour populations are ordered from light to dark?

Very minor thing but should the coastlines borders be the same thickness and colour as the country borders?

cclarkson commented:
@alimanfoo , @hardingnj , @EricRLucas 

Ready for a review/merge I think.

Cheers

hardingnj commented:
Looks great! A couple of thoughts:

- Should it be *Populations* in the key instead of *Population*?

- I think the use of arrows from the circle to the collection site is a little inconsistent. If no arrow I assume it's centred on the location? I think I'd prefer arrows on all, in case of Cameroon etc, it obstructs the location habitat. Maybe the arrows are too similar to the border lines as well.

And this is very much just thinking aloud; especially with phase 3 upcoming:

- Do we need even colours here? Linking the cicles to the legend is done only by color at the moment... what about colouring by gam/col, size as is. Then using a letter to denote the population?

- Should we add the number of collection sites to the legend?

alimanfoo commented:
Thanks Chris, Nick. Responding to some comments...

> I think the use of arrows from the circle to the collection site is a little inconsistent. If no arrow I assume it's centred on the location? I think I'd prefer arrows on all, in case of Cameroon etc, it obstructs the location habitat. 

This is probably a good point, would give a more precise indication by eye of where sampling was done for all populations.

> Maybe the arrows are too similar to the border lines as well.

Personally not too concerned about this, although maybe if both country and coastline borders were slightly thinner, then the pie and arrow/line edges could be slightly thicker?

> Do we need even colours here? Linking the cicles to the legend is done only by color at the moment... what about colouring by gam/col, size as is. Then using a letter to denote the population?

Mulling this over...

Potentially a good point, the links between the labels in the plot (e.g., "GHcol") and the labels in the legend (e.g., "Ghana coluzzii") are not explicit, only via colour, which isn't perfect. Maybe we could do away with the population legend altogether, and instead we could expand the abbreviations and give sample sizes in the figure caption text (e.g., "GHcol = Ghana An. coluzzii, n=55; etc.")?

Also I take the point that there is a lot of colour going on, and not giving any actual information in this plot. Maybe colouring by species (gam/col/uncertain) is better as then colour is conveying some actual information?

> Should we add the number of collection sites to the legend?

FWIW don't think that's necessary, all details of sites will be in methods.

hardingnj commented:
Re: colour. I think it does orientate the reader for the upcoming plots, which is helpful, but I do worry there is a lot of colour for not a lot of additional information.

alimanfoo commented:
Proposal:
* Colour by species (gambiae = blue, coluzzii = red, unknown = ?)
* Remove population legend
* Add species colour legend
* Add arrows for all circles
* Keep each population as a separate circle, include the population label (e.g., "GNgam")
* Scale circle to number of samples by area

Text caption expands each population, e.g., "CIcol = Cote d'Ivoire *An. coluzzii*; GHcol = Ghana *An. coluzzii*, n=55; etc."

hardingnj commented:
Just dashing off now- but first impressions are that it looks great! I'll review properly tomorrow

EricRLucas commented:
Hi guys,

Sorry for the silence. Still figuring out how GitHub works. Map looks great, and I don't have any suggestions to add. Thanks Chris! Now to work out how to submit my code for the Dxy plots.

cclarkson commented:
Oops, colours are the correct way around now...

cclarkson commented:
@alimanfoo @hardingnj @EricRLucas 

New legend - more informative, fills ugly white space better.

cclarkson commented:
@alimanfoo @hardingnj @EricRLucas 

Gabon moved into the sea and legend title altered.
